compiler/rustc_data_structures/src/sip128.rs

```diff
@@ -1,21 +1,31 @@
 //! This is a copy of `core::hash::sip` adapted to providing 128 bit hashes.
 
-use std::cmp;
 use std::hash::Hasher;
-use std::mem;
+use std::mem::{self, MaybeUninit};
 use std::ptr;
 
 #[cfg(test)]
 mod tests;
 
+const ELEM_SIZE: usize = mem::size_of::<u64>();
```

Comment:
Comments about the choices here would be good. I guess `ELEM_SIZE` matches the word size on 64-bit platforms, but it's unclear to me if the value of 8 for `BUFFER_SIZE_ELEMS` has any particular significance.